cache/s3proxy/s3proxy.go

```diff
@@ -196,6 +199,22 @@ func (c *s3Cache) Put(ctx context.Context, kind cache.EntryKind, hash string, si
 	}
 }
 
+func (c *s3Cache) UpdateModificationTimestamp(ctx context.Context, bucket string, object string) {
+	src := minio.CopySrcOptions{
+		Bucket: bucket,
+		Object: object,
+	}
+
+	dst := minio.CopyDestOptions{
+		Bucket: bucket,
+		Object: object,
+	}
+
+	_, err := c.mcore.ComposeObject(context.Background(), dst, src)
```

Comment:
I see that the copying is done server side- nice. Are there any downsides to having this always on?

Reply:
Good question - I don't see any downsides in terms of performance or storage usage but I'm running self hosted s3 storage. As I understand AWS s3 [pricing documentation](https://aws.amazon.com/s3/pricing/) there is some cost for coping object - `you pay the S3 charges for storage in the selected destination S3 storage classes, for the primary copy, for replication PUT request`. So making this turn on by default could potentially generate costs for people using AWS so maybe it's better to leave it as opt-in?

Reply:
OK, let's make it opt-in to be on the safe side.